pkg/kn/commands/types.go

```diff
@@ -16,28 +16,38 @@ package commands
 
 import (
 	"io"
+	"os"
+	"path/filepath"
 
 	serving "github.com/knative/serving/pkg/client/clientset/versioned/typed/serving/v1alpha1"
+	homedir "github.com/mitchellh/go-homedir"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-// KubeCfgFile is the path for the Kubernetes config
-var KubeCfgFile string
-
 // Parameters for creating commands. Useful for inserting mocks for testing.
 type KnParams struct {
-	Output         io.Writer
-	ServingFactory func() (serving.ServingV1alpha1Interface, error)
+	Output           io.Writer
+	ServingFactory   func() (serving.ServingV1alpha1Interface, error)
+	CurrentNamespace string
+	KubeCfgFile      string
 }
 
-func (c *KnParams) Initialize() {
+func (c *KnParams) Initialize() error {
 	if c.ServingFactory == nil {
-		c.ServingFactory = GetConfig
+		c.ServingFactory = c.GetConfig
 	}
+	var err error
+	c.CurrentNamespace, err = c.GetCurrentNamespace()
+	return err
 }
 
-func GetConfig() (serving.ServingV1alpha1Interface, error) {
-	config, err := clientcmd.BuildConfigFromFlags("", KubeCfgFile)
+func (c *KnParams) GetConfig() (serving.ServingV1alpha1Interface, error) {
+	if c.KubeCfgFile == "" {
+		if err := c.initKubeConfig(); err != nil {
+			return nil, err
+		}
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", c.KubeCfgFile)
```

Comment:
we should also get `ClientConfig()` from `clientcmd.NewNonInteractiveDeferredLoadingClientConfig()` instead of doing `initKubeConfig()`. i dont think we actually need initKubeConfig at all.